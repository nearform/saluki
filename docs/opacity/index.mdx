import { Opacity } from './'

# Opacity

---

## Example

<Opacity>Hello</Opacity>

```js
const Text = styled.h1`
  ${opacity.half}
`
```

## Rules

```js
opacity.full
opacity.half
opacity.none
opacity.inherit
```

## Configuration

Configurable via;

```
opacity
```

## Defaults

```js
{
  full: 1,
  half: 0.5,
  none: 0,
  inherit: 'inherit'
}
```
