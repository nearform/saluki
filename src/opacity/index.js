export const defaultRules = {
  full: 1,
  half: 0.5,
  none: 0,
  inherit: 'inherit'
}

function generateRules(rules) {
  const newRules = {}

  for (const [key, value] of Object.entries(rules)) {
    newRules[key] = {
      opacity: value
    }
  }

  return newRules
}

export default function(opacity) {
  return generateRules(opacity)
}
