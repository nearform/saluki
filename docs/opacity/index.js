import styled from 'styled-components/macro'
import { opacity } from '../../src'

export const Opacity = styled.h1`
  ${opacity.half}
`
