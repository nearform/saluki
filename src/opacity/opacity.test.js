import opacity, { defaultRules } from './'

describe('opacity', () => {
  describe('when no custom config is provided', () => {
    it('should return the default config', () => {
      expect(opacity(defaultRules)).toMatchSnapshot()
    })
  })

  describe('when custom config is provided', () => {
    describe('when the custom config overrides a value', () => {
      it('should replace the default value', () => {
        const combinedRules = {
          ...defaultRules,
          ...{ full: 0.99 }
        }
        expect(opacity(combinedRules)).toMatchSnapshot()
      })
    })

    describe('when the custom config provides a new value', () => {
      it('should add the value', () => {
        const combinedRules = {
          ...defaultRules,
          ...{ quarter: 0.25 }
        }
        expect(opacity(combinedRules)).toMatchSnapshot()
      })
    })
  })
})
