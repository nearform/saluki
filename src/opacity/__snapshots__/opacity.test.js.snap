// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`opacity when custom config is provided when the custom config overrides a value should replace the default value 1`] = `
Object {
  "full": Object {
    "opacity": 0.99,
  },
  "half": Object {
    "opacity": 0.5,
  },
  "inherit": Object {
    "opacity": "inherit",
  },
  "none": Object {
    "opacity": 0,
  },
}
`;

exports[`opacity when custom config is provided when the custom config provides a new value should add the value 1`] = `
Object {
  "full": Object {
    "opacity": 1,
  },
  "half": Object {
    "opacity": 0.5,
  },
  "inherit": Object {
    "opacity": "inherit",
  },
  "none": Object {
    "opacity": 0,
  },
  "quarter": Object {
    "opacity": 0.25,
  },
}
`;

exports[`opacity when no custom config is provided should return the default config 1`] = `
Object {
  "full": Object {
    "opacity": 1,
  },
  "half": Object {
    "opacity": 0.5,
  },
  "inherit": Object {
    "opacity": "inherit",
  },
  "none": Object {
    "opacity": 0,
  },
}
`;
